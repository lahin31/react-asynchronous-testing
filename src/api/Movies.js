import axios from 'axios'

export default async () => {
  const response = await axios.get('https://api.themoviedb.org/3/movie/popular?api_key=1b5adf76a72a13bad99b8fc0c68cb085&language=en-US')
  return response.data.results
}