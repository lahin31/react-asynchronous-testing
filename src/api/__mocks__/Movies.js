const fakeData = [
  {
    id: 1,
    title: "Test Title",
    poster_path: "/test.jpg",
    release_date: "2019-01-03",
    popularity: 1056,
    vote_average: 8.6
  }
]

export default async () => {
  return await new Promise(resolve => {
    resolve(fakeData)
  })
}