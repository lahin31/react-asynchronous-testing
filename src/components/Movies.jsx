import React, { Component } from 'react';
import MoviesApi from '../api/Movies'
import Movie from './Movie'
import "./Movies.css"

class Movies extends Component {
  state = { 
    movies: [],
    error: ""
  }
  componentDidMount() {
    this.fetchMovies()
  }

  fetchMovies = async () => {
    try {
      const data= await MoviesApi()
      this.setState({ movies: data.slice(0, 5) })
    } catch(err) {
      this.setState({
        error: err
      })
    }
  }
  render() { 
    const { movies } = this.state;
    return (
      <div className="movies_container">
        { movies.map(movie => {
          return (
            <Movie movie={movie} key={movie.id} />
          )
        })}
      </div>
    )
  }
}
 
export default Movies;