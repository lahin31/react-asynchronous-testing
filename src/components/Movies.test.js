import Movies from "./Movies"

jest.mock("../api/Movies")

describe("Movies Component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Movies />) // shallow - returns a light-weight representation of what component renders
    expect(wrapper).toMatchSnapshot();
  })

  it("fetches movies from movieDB api and renders them on mount", done => {
    const wrapper = shallow(<Movies />)

    setTimeout(() => {
      wrapper.update()
      const state = wrapper.state()
      expect(state.error).toEqual("")
      expect(wrapper.find("Movie").length).toEqual(1)
      done()
    })
  })

})