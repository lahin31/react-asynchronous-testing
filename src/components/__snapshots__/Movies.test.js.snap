// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies Component renders correctly 1`] = `
<div
  className="movies_container"
/>
`;
